packages/playwright-ct-core/src/tsxTransform.ts

```diff
@@ -71,19 +73,35 @@ export default declare((api: BabelAPI) => {
         if (!t.isStringLiteral(importNode.source))
           return;
 
-        let components = 0;
+        const ext = path.extname(importNode.source.value);
+
+        // Convert all non-JS imports into refs.
+        if (!allJsExtensions.has(ext)) {
+          for (const specifier of importNode.specifiers) {
+            if (t.isImportNamespaceSpecifier(specifier))
+              continue;
+            const { localName, info } = importInfo(importNode, specifier, this.filename!);
+            importInfos.set(localName, info);
+          }
+          p.skip();
+          p.remove();
+          return;
+        }
+
+        // Convert JS imports into refs only if they are used by JSX.
```

Comment:
It took me awhile to understand this comment. I'd recommend `Convert JS imports that are used as components in JSX expressions into refs`.